adafruit_fram.py

```diff
@@ -93,103 +108,58 @@ def write_protected(self):
             When no ``WP`` pin is supplied, protection is only at the software
             level in this library.
         """
-        if not self._wp_pin is None:
-            status = self._wp_pin.value
-        else:
-            status = self._wp
-        return status
+        return self._wp if self._wp_pin is None else self._wp_pin.value
 
     @write_protected.setter
     def write_protected(self, value):
         self._wp = value
         if not self._wp_pin is None:
             self._wp_pin.value = value
 
-    def write_protect_pin(self, wp_pin, write_protect=False):
-        """ Assigns the write protection (``WP``) pin.
+    def __getitem__(self, key):
+        if isinstance(key, int):
+            if key > self._max_size:
+                raise ValueError("Register '{0}' greater than maximum FRAM size."
+                                 " ({1})".format(key, self._max_size))
+            return self._read_byte(key)
+        elif isinstance(key, slice):
+            registers = list(range(key.start if not key.start is None else 0,
+                             key.stop if not key.stop is None else self._max_size,
+                             key.step if not key.step is None else 1))
+            if (registers[0] + len(registers)) > self._max_size:
+                raise ValueError("Register + Length greater than maximum FRAM size."
+                                 " ({0})".format(self._max_size))
+
+            read_buffer = bytearray(len(registers))
+            for i, register in enumerate(registers):
+                read_buffer[i] = self._read_byte(register)[0]
+            return read_buffer
+
+    def __setitem__(self, key, value):
+        if self.write_protected:
+            raise RuntimeError("FRAM currently write protected.")
 
-        :param: wp_pin: The ``board.PIN`` object connected to the ``WP`` pin
-                        on the breakout board/chip. To remove a previously
-                        set ``WP`` pin, set this value to ``None``.
-        :param: bool write_protect: Turn on/off write protection immediately
-                                    when setting the pin. Default is ``False``
+        if isinstance(key, int):
+            if not isinstance(value, int):
+                raise ValueError("Data must be an integer.")
+            if key.start > self._max_size:
```

Comment:
Looks like slice code? Trying to use `.start` on an `int` type.

Reply:
hmm...swear I fixed that on the `fix the fixes` commit.